src/transaction/log_manager.c

```diff
@@ -4807,11 +4849,17 @@ log_commit_local (THREAD_ENTRY * thread_p, LOG_TDES * tdes, bool retain_lock, bo
 	{
 	  LOG_LSA commit_lsa;
 
-	  if (LSA_ISNULL (&tdes->posp_nxlsa))
+	  /* To write unlock log before releasing locks for transactional consistencies. When a transaction(T2) which
+	   * is resumed by this committing transaction(T1) commits and a crash happens before T1 completes, transaction
+	   * consistencies will be broken because T1 will be aborted during restart recovery and T2 was already
+	   * committed. */
+	  /* TODO - with GC, no need to append log commit or ha commit. */
+	  if (!LOG_CHECK_LOG_APPLIER (thread_p) && tdes->is_active_worker_transaction ()
+	      && log_does_allow_replication () == true)
 	    {
 	      tx_group group;
 	      group.add (tdes->tran_index, 0, tdes->state);
-	      log_append_group_complete (thread_p, tdes, 0, group, &commit_lsa);
+	      log_append_group_complete (thread_p, tdes, 0, group, &commit_lsa, NULL);
```

Comment:
add TODO : this should be removed when complete manager is functional

Reply:
Already removed in another patch that waits for CBRD-22944.